app/javascript/packages/store/src/graphql/client.ts

```diff
@@ -37,16 +37,20 @@ const graphql = (query, variables, callbacks) => {
       if (isObject(errors) && !isEmpty(errors)) {
         // const errors = data[Object.keys(data)[0]];
         // callbacks['error'] ? callbacks['error'](res, errors['errors']) : null
-        if (
-          (errors[0].extensions &&
-            errors[0].extensions.code === 'unauthorized') ||
-          errors[0]?.code === 'unauthenticated'
-        ) {
-          //@ts-ignore
-          store.dispatch(errorMessage(errors[0].message));
-          store.dispatch(doSignout());
-          //@ts-ignore
-          return store.dispatch(clearCurrentUser());
+        if (errors[0].extensions) {
+          if (errors[0]?.code === UNAUTHENTICATED) {
```

Comment:
here I think we need to add the unauthorized check too

Reply:
Oh this part checks for unauthenticated to log the user out. The else statement below it handles the unauthorized error to show the access denied error